File: examples/python/weather_agent/model/weather.py
"""Typed weather contracts shared by the agent and tool modules."""

from dataclasses import dataclass


@dataclass(frozen=True, slots=True)
class WeatherResult:
    """Structured observation returned by the weather tool."""

    city: str
    temperature_c: int


@dataclass(frozen=True, slots=True)
class WeatherAnswer:
    """Structured terminal output required from the model."""

    city: str
    temperature_c: int
    summary: str

File: examples/javascript/weather-agent/src/model/weather.ts
import {z} from "../../../../../sdks/javascript/dist/index.js";

/** Runtime schema for model-produced weather tool arguments. */
export const WeatherArguments = z.object({city: z.string().min(1)});

/** Runtime schema for the application tool result. */
export const WeatherResult = z.object({
  city: z.string(),
  temperature_c: z.number().int(),
});

/** Runtime schema for the model's terminal structured output. */
export const WeatherAnswer = z.object({
  city: z.string(),
  temperature_c: z.number().int(),
  summary: z.string(),
});

/** Typed terminal answer inferred from the runtime schema. */
export type WeatherAnswerValue = z.infer<typeof WeatherAnswer>;

/** Typed result inferred from the runtime schema. */
export type WeatherResultValue = z.infer<typeof WeatherResult>;
